ui/desktop/electron-app/src/helpers/app-updater.js

```diff
@@ -19,6 +20,33 @@ if (debug && process.env.APP_UPDATER_CURRENT_VERSION) {
   currentVersion = process.env.APP_UPDATER_CURRENT_VERSION;
 }
 
+const returnArchitectureToUpdate = () => {
+  const nodeArchitecture = process.arch;
+
+  if (nodeArchitecture === 'arm64') {
+    return nodeArchitecture;
+  }
+
+  if (nodeArchitecture === 'amd64') {
+    const runsOnRosetta = exec(
```

Comment:
We should use `execSync` or `spawnSync` rather than `exec`, that simplifies it a bit more.

You're returning the child process and using that as the check instead of the output. You would have to check what the output is returning (and then based on the output determine if we're running in rosetta).

Keep in mind that outputs are buffers by default FYI